rotkehlchen/chain/evm/decoding/curve/curve_cache.py

```diff
@@ -106,6 +104,147 @@ def read_curve_pools_and_gauges(chain_id: ChainID) -> tuple[dict[ChecksumEvmAddr
     return curve_pools, curve_gauges
 
 
+def _ensure_single_pool_curve_tokens_existence(
+        evm_inquirer: 'EvmNodeInquirer',
+        pool: CurvePoolData,
+) -> CurvePoolData | None:
+    encounter = TokenEncounterInfo(
+        description=f'Querying curve pools for {evm_inquirer.chain_id.to_name()}',
+        should_notify=False,
+    )
+    # Ensure pool coins exist in the globaldb.
+    # We have to create underlying tokens only if pool utilizes them.
+    if pool.underlying_coins is None or pool.underlying_coins == pool.coins:
+        # If underlying coins is None, it means that pool doesn't utilize them.
```

Comment:
I have forgotten what is the difference between `underlying_coins` and coins. Could you document it in the data structure?